### PR TITLE
Milestone 4

### PR DESCRIPTION
**Pull request by MarvellousUbani**
- Created Comments Functionality
- Created Likes Functionality
- Wrote Tests 
- Created UI's for both features

### CODE REVIEW

**Comment by samgaco** [comment]
Hi @MarvellousUbani and @Ashah15 :high_brightness: 

- [ ] The users should be able to like posts for this milestone

- [ ] Write tests for the like and comment models


Please work on that and submit back for another review.

**Comment by MarvellousUbani** [comment]
> Hi @MarvellousUbani and @Ashah15 
> 
> * [ ]  The users should be able to like posts for this milestone
> * [ ]  Write tests for the like and comment models
> 
> Please work on that and submit back for another review.
> 
> * ###### STATUS OF THE MILESTONE: REQUESTED CHANGES stop_sign

Hi and I hope you are doing great, 

Yes the Users are able to like comments when they click on the thumbs up or thumbs down icon.
Also we added tests for likes and comments on the spec/feature folder

File: app/views/posts/_post.html.erb [hunk 1]
```diff
@@ -13,10 +13,36 @@
 <p class="post-time-ago my-2"><%= time_ago_in_words(post.created_at) %> ago.</p>
 <p class="post-content"><%= post.content %></p>
 <div class="d-flex justify-content-between post-like-line">
-	<button class="trans-btn"><i class="fa fa-thumbs-o-up" aria-hidden="true"></i>  Like</button>
+	<ul class="d-flex pl-0">
+		<li>
+			<% unless current_user.likes.exists?(post_id: post.id) %>
```
Comment (wathigo) [inline app/views/posts/_post.html.erb]: **[REQUIRED]**
* View-related logic can be encapsulated in View helpers.
In this case we could have a method in `app/helpers/posts_heper` with this method.
```
module PostsHelper  
  def liked?(post_id)
    current_user.likes.exists?
  end  
end
```
and then just call the method in the views like
`<% unless liked?(post.id) %>`

([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright)

File: app/views/posts/_post.html.erb [hunk 2]
```diff
@@ -13,10 +13,36 @@
...
+			  <%= button_to "👍",{:controller => 'likes', :action => 'create',:user_id => current_user.id, :post_id => post.id}, {:method => :post, :class => "trans-btn font-red mid-font"} %>
+			<% else %>
+			  <%= button_to "👎",{:action => 'destroy',
+			  	:controller => 'likes',:id => post.likes.find_by(user_id:current_user.id).id },
```
Comment (wathigo) [inline app/views/posts/_post.html.erb]: **[REQUIRED]**
Let's move all the view related logic to the helpers. Methods in `app/helpers` are automatically available to the views.
([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright).

**More Resources** 
* [Where Do I Put My Code?](http://codefol.io/posts/Where-Do-I-Put-My-Code).
* [Beginners Guide to Rails helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers)

File: spec/features/create_comment.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+RSpec.describe 'User comments', type: :feature do
+  before :each do
+    User.create(name: 'mike', email: 'mike@gmail.com', password: '111111')
+    Post.create(content: 'text', author_id: 1)
+  end
+
+  scenario 'has comment content' do
+    login_with 'mike@gmail.com', '111111'
+    fill_in 'comment_content', with: 'Good Day'
```
Comment (wathigo) [inline spec/features/create_comment.rb]: **[REQUIRED]**
This line is causing this error while running the tests.
```
Failure/Error: fill_in 'comment_content', with: 'Good Day'
     
     Capybara::ElementNotFound:
       Unable to find field "comment_content" that is not disabled
```

File: spec/features/likes.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+RSpec.describe 'User like', type: :feature do
+  before :each do
+    User.create(name: 'mike', email: 'mike@gmail.com', password: '111111')
+    Post.create(content: 'text', author_id: 1)
+  end
+
+  scenario 'has like' do
+    login_with 'mike@gmail.com', '111111'
+    click_on 'Like'
```
Comment (wathigo) [inline spec/features/likes.rb]: **[REQUIRED]**
This line is causing this :point_down:  error.
```
Failure/Error: click_on 'Like'
     
     Capybara::ElementNotFound:
       Unable to find link or button "Like"
```

File: app/views/posts/_post.html.erb (hunk 1 above)
Comment (MarvellousUbani) [reply app/views/posts/_post.html.erb]: Thanks @wathigo
We fixed this and resent the pull request for review

File: app/views/posts/_post.html.erb (hunk 2 above)
Comment (MarvellousUbani) [reply app/views/posts/_post.html.erb]: We also put the view related logic `post.likes.find_by(user_id:current_user.id).id }` into a helper method

File: spec/features/create_comment.rb (hunk 3 above)
Comment (MarvellousUbani) [reply spec/features/create_comment.rb]: We resolved this as well

File: spec/features/likes.rb (hunk 4 above)
Comment (MarvellousUbani) [reply spec/features/likes.rb]: We have also resolved this.

**Pull request merged**